crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_literal_union.rs

```diff
@@ -13,30 +13,28 @@ use crate::checkers::ast::Checker;
 use crate::fix::snippet::SourceCodeSnippet;
 
 /// ## What it does
-/// Checks for the presence of redundant `Literal` types and builtin super
-/// types in an union.
+/// Checks for redundant unions between a `Literal` and a builtin supertype of
+/// that `Literal`.
 ///
 /// ## Why is this bad?
-/// The use of `Literal` types in a union with the builtin super type of one of
-/// its literal members is redundant, as the super type is strictly more
-/// general than the `Literal` type.
-///
+/// Using a `Literal` type in a union with its builtin supertype is redundant,
+/// as the supertype will be strictly more general than the `Literal` type.
 /// For example, `Literal["A"] | str` is equivalent to `str`, and
-/// `Literal[1] | int` is equivalent to `int`, as `str` and `int` are the super
-/// types of `"A"` and `1` respectively.
+/// `Literal[1] | int` is equivalent to `int`, as `str` and `int` are the
+/// supertypes of `"A"` and `1` respectively.
 ///
 /// ## Example
 /// ```python
 /// from typing import Literal
 ///
-/// A: Literal["A"] | str
+/// x: Literal["A", b"B"] | str
 /// ```
 ///
 /// Use instead:
 /// ```python
 /// from typing import Literal
 ///
-/// A: Literal["A"]
+/// x: Literal["A"] | str
```

Comment:
Shouldn't this be `str`?

> For example, `Literal["A"] | str` is equivalent to `str`, and

Reply:
Whoops, good catch. This should be `Literal[b"B"] | str`.